### PR TITLE
Mono | Use `dma` deployment config addresses as source of truth for `addresses` package

### PR DESCRIPTION
**Pull request by zerotucks**
Note: Please read **Relevant Files** section before reviewing. Most files are simple import changes and can be ignored.

**Changes**
* Remove old address json files
* Convert @oasisdex/addresses into a package /src directory builds to /lib
* Updated all ADDRESSES imports to rely on source code not built version (@oasisdex/addresses/src) // many files
* Cleaned up network configs so they conform to SystemConfig type

**Relevant Files**
* packages/addresses/src/index.ts
* packages/dma-contracts/test/addresses/mainnet.ts // shows example of ADDRESSES record being used.
* packages/dma-deployments/configs/goerli.conf.ts // shows how configs have been typed appropriately
* packages/dma-deployments/configs/mainnet.conf.ts // as above

### CODE REVIEW

File: packages/addresses/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+import { DeploymentConfig, SystemKeys } from '@oasisdex/dma-deployments/types/deployment-config'
+import { Network } from '@oasisdex/dma-common/utils/network'
+
+type DeployedNetworks = Network.MAINNET | Network.OPT_MAINNET | Network.GOERLI
+// TODO: Make address entries type safe after https://github.com/OasisDEX/oasis-earn-sc/pull/249/files is merged
+export type Addresses = Record<DeployedNetworks, Record<SystemKeys, any>>
+
+export const ADDRESSES: Addresses = {
+  [Network.MAINNET]: {
```
Comment (marcinciarka) [inline packages/addresses/src/index.ts:13]: What do you think about exporting `[Network.MAINNET]` again but as `[Network.HARDHAT]`? Just for the ease of use.

File: packages/addresses/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      },
+      v3: {
+        ...extractAddressesFromConfig(mainnetConfig.aave.v3),
+      },
+    },
+    maker: {
+      ...extractAddressesFromConfig(mainnetConfig.maker),
+    },
+  },
+  [Network.OPT_MAINNET]: {
```
Comment (marcinciarka) [inline packages/addresses/src/index.ts:37]: Can we have a full name `OPTIMISM_MAINNET`? OPT sounds like 'optional'.

Comment (zerotucks) [reply packages/addresses/src/index.ts:37]: It's been updated in another PR -> https://github.com/OasisDEX/oasis-earn-sc/pull/249

File: packages/addresses/src/index.ts (hunk 1 above)
Comment (zerotucks) [reply packages/addresses/src/index.ts:13]: But Hardhat could be using Optimism or Mainnet addresses depending on the fork

File: packages/addresses/package.json [hunk 3]
```diff
@@ -1,9 +1,17 @@
...
   "version": "1.0.0",
-  "main": "index.ts",
+  "main": "lib/src/index.js",
+  "typings": "lib/src/index.d.ts",
   "license": "MIT",
+  "scripts": {
+    "build": "yarn tsc -b",
+    "watch": "yarn tsc -b -w",
+    "lint": "yarn tsc -p ./tsconfig.json --noEmit",
+    "prepublish": "npm run build"
```
Comment (jakubswierczek) [inline packages/addresses/package.json]: We are using `yarn` so it should be `yarn build`

File: packages/addresses/src/index.ts (hunk 2 above)
Comment (marcinciarka) [reply packages/addresses/src/index.ts:37]: Nice.

File: packages/addresses/src/index.ts (hunk 1 above)
Comment (marcinciarka) [reply packages/addresses/src/index.ts:13]: Right.

**Review by marcinciarka — APPROVED** [review]

File: packages/dma-common/utils/maker/vault.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { JsonRpcProvider } from '@ethersproject/providers'
-import { ADDRESSES } from '@oasisdex/addresses'
+import { ADDRESSES } from '@oasisdex/addresses/src'
```
Comment (jakubswierczek) [inline packages/dma-common/utils/maker/vault.ts]: Can we have only `@oasisdex/addresses` imports?

File: packages/dma-common/utils/swap/1inch.ts [hunk 5]
```diff
@@ -1,4 +1,5 @@
-import { ADDRESSES } from '@oasisdex/addresses'
+import { ADDRESSES } from '@oasisdex/addresses/src'
```
Comment (jakubswierczek) [inline packages/dma-common/utils/swap/1inch.ts]: Import stuff

File: packages/dma-common/utils/swap/1inch.ts [hunk 6]
```diff
@@ -1,4 +1,5 @@
...
+import { Network } from '@utils/network'
```
Comment (jakubswierczek) [inline packages/dma-common/utils/swap/1inch.ts]: Is `@utils` a separate package? It looks like. Maybe for internal imports, we should use `~` for example.

**Review by jakubswierczek — CHANGES_REQUESTED** [review]

**Pull request merged**